pkg/ipamd/ipamd.go

```diff
@@ -502,8 +508,12 @@ func (c *IPAMContext) getLocalPodsWithRetry() ([]*k8sapi.K8SPodInfo, error) {
 func (c *IPAMContext) StartNodeIPPoolManager() {
 	sleepDuration := ipPoolMonitorInterval / 2
 	for {
-		time.Sleep(sleepDuration)
-		c.updateIPPoolIfRequired()
+		waitLongEnough := waitLongEnoughSinceLastAction(c.lastNodeIPPoolAction)
+		// TODO: adding another condition of is_new_ENI_available
+		if !c.disableENIProvisioning && waitLongEnough {
```

Comment:
I don't think this is needed. The check for new ENIs happens in `c.nodeIPPoolReconcile`